app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,20 +72,142 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfFirstNameEditTextIsInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
+    public void signIn_checkIfLastInitialEditTextIsInitiallyEmpty() {
+        onView(withId(last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+    }
+
+    @Test
+    public void signIn_chooseAvatar_01() {
```

Comment:
~~Remove signIn_ prefix from the tests. The syntax should be chooseAvatarXX.
Since this is an adapter backed view, you can make use of onData.~~
Each test is performing multiple checks and they're fragile due to redundant code.
Make sure to check the amount of displayed avatars matches the actual avatars available.